doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,339 @@
+     - tree -psv /sys_switch/
+3. Porting S3IP sysfs framework to platform project
+ - Create the S3IP sysfs service configuration file. 
+     - Modify /s3ip/s3ip_sysfs_service/s3ip_sysfs_conf.json to create a configuration file
+     - This file should be installed to /etc/s3ip/s3ip_sysfs_conf.json on the device
+ - Implement the platform hardware driver, and export parameters through s3ip_sysfs.ko
+     - s3ip_sysfs. ko is generated by compiling /s3ip/s3ip_sysfs_frame source code
+ - Install the s3ip-sysfs.service on the device
+    - Copy service file /s3ip/scripts/s3ip-sysfs.service to the project.
+    - The service file needs to be installed in /etc/systemd/system/s3ip-sysfs.service on the device.
+    - When the platform package is installed, the vendor should enable s3ip-sysfs.service and start this service automatically when the device reboot
+    - After the platform package is uninstalled, the vendor should disable the s3ip-sysfs.service and remove the service file
+4. Self-test after integration
+   - Ensure that the /sys_switch directory conforms to the S3IP sysfs specification 
+   - Hardware property information can be read and written normally
+
+### 4 Compare with [PDDF](https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md)
+
+The S3IP sysfs specification is intended to provide a more general set of hardware access interfaces. Using sysfs is definitely the best option, as it fits into the Linux design philosophy of everything being a file, and there are more tools and languages to deal with it.
+
+For traditional driver engineers, The S3IP SYSFS interface is very friendly and easy to debug, they only need to care about the content and format of the sysfs node. So the learning cost is zero and it's easy to understand and implement. 
+
+For ODM vendors new to SONIC, opt for PDDF because you can do platform adaptation with a configuration file, which is cool and popular.
+
+S3IP SYSFS and PDDF both share similar goals of reducing vendor adaptation effort and increasing the reusability of device management software.
+
+However, there are many differences between the two frameworks, which are listed below. ODM vendors can choose the framework based on actual requirements.
+
+| Item  | S3IP sysfs|PDDF|
+|-|-|-|
+|Requirements| The requirements are put forward from the user's perspective, and the sysfs node is summarized from the actual operation experience. The goal is to unify the interface for device management | The requirements are proposed from the technical point of view.TBD
+|Ecological | Devices compliant with S3IP SYSFS specification have been widely used in data centers.<br> The [S3IP](http://www.s3ip.org/) project involved both vendors and users[S3IP] (including Tencent, Alibaba, Baidu, Kuaishou, Meituan, Jingdong and more than a dozen ODM vendors). Vendors and users complete a closed-loop of requirements. standards and debugging tools that have the ability to iterate continuously. | TBD

```

Comment:
Ecological -> Ecosystem is probably what you meant.

Reply:
ok，accept

Reply:
EcoSystem: (PDDF) PDDF is a new framework and it is developed in the SONiC context. Some ODM platforms are already using PDDF. PDDF is an underlying framework which ODMs can use for faster development but it does not exposes any fixed SysFS nodes to the user.

Reply:
Please add this comment in the PDDF column for the row "EcoSystem"

Reply:
already updated